syntax/pb_test.go

```diff
@@ -0,0 +1,40 @@
+package syntax
+
+import (
+	"testing"
+
+	"github.com/arr-ai/arrai/rel"
+	"github.com/arr-ai/wbnf/parser"
+	"github.com/stretchr/testify/assert"
+)
+
+func TestTransformProtoBufToTupleFlow(t *testing.T) {
+	code := "//encoding.proto.decode(//os.file('../pb/test/sysl.pb'), //os.file('../pb/test/petshop'), 'Module')"
+	pc := ParseContext{SourceDir: ".."}
+	ast, err := pc.Parse(parser.NewScanner(code))
+	assert.NoError(t, err)
+
+	codeExpr := pc.CompileExpr(ast)
+	val, err := codeExpr.Eval(rel.EmptyScope)
+
+	assert.NoError(t, err)
+	tuple, _ := val.(rel.Tuple)
+	apps, _ := tuple.Get("apps")
+
+	assert.NotNil(t, apps)
+}
+
+func TestTransformProtoBufToTupleCompareResult(t *testing.T) {
+	code := "//encoding.proto.decode(//os.file('../pb/test/sysl.pb'), //os.file('../pb/test/petshop'), 'Module')"
```

Comment:
Do you want decoding them to `rel.Value`? The decoding of protobuf bytes is `FileDescriptor`, and if want to convert it to an arr.ai type `rel.Value`. It has to add a wrapper which implements interface `rel.Value` and it looks like:
```
type FileDescriptorWrapper struct {
	descriptor pr.FileDescriptor
}

func (descriptor FileDescriptorWrapper) Kind() int {
	return 1
}

func (descriptor FileDescriptorWrapper) Equal(desc interface{}) bool {
	return false
}

func (descriptor FileDescriptorWrapper) Eval(scope rel.Scope) (rel.Value, error) {
	return nil, nil
}

func (descriptor FileDescriptorWrapper) Export() interface{} {
	return nil
}

func (descriptor FileDescriptorWrapper) Hash(seed uintptr) uintptr {
	return 0
}

func (descriptor FileDescriptorWrapper) IsTrue() bool {
	return false
}

func (descriptor FileDescriptorWrapper) Less(v rel.Value) bool {
	return false
}

func (descriptor FileDescriptorWrapper) Negate() rel.Value {
	return nil
}

func (descriptor FileDescriptorWrapper) Source() parser.Scanner {
	return parser.Scanner{}
}

func (descriptor FileDescriptorWrapper) String() string {
	return ""
}
```
Most of them will be empty methods, and it looks odd. Also it has to change some other methods, for example:
```
func reprValue(v Value, w io.Writer) {
    switch v := v.(type) {
    case String:
        reprString(v, w)
    case Bytes:
        reprBytes(v, w)
    case Array:
        reprArray(v, w)
    case Dict:
        reprDict(v, w)
    case GenericSet:
        reprSet(v, w)
    case StringCharTuple:
        reprStringCharTuple(v, w)
    case ArrayItemTuple:
        reprArrayItemTuple(v, w)
    case DictEntryTuple:
        reprDictEntryTuple(v, w)
    case *GenericTuple:
        reprTuple(v, w)
    case Number:
        reprNumber(v, w)
    default:
        panic(fmt.Errorf("Repr(): unexpected Value type %T: %[1]v", v)) //nolint:golint
    }
}
```
I think this will add unnecessary complexity.